src/lib/KeyStoreIndexeddb.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Usage:
+ * <script src="lib/key.js"></script>
+ * <script src="lib/key-store-indexeddb.js"></script>
+ * const keyStore = KeyStore.instance;
+ * const accounts = await keyStore.list();
+ */
+class KeyStore {
+
+    static get instance() {
+        /** @type {KeyStore} */
+        this._instance = this._instance || new KeyStore();
+        return this._instance;
+    }
+
+    constructor() {
+        /** @type {IDBDatabase} */
+        this._db;
+        this._connected = false;
+    }
+
+    /**
+     * @returns {Promise.<IDBDatabase>}
+     * @private
+     */
+    connect() {
+        if (this._connected && this._db) return Promise.resolve(this._db);
+
+        return new Promise((resolve, reject) => {
+            const request = self.indexedDB.open(KeyStore.DB_NAME, KeyStore.DB_VERSION);
+
+            request.onsuccess = () => {
+                this._db = request.result;
+                this._connected = true;
+                resolve(this._db);
+            };
+
+            request.onerror = () => reject(request.error);
+
+            request.onupgradeneeded = event => {
+                /** @type {IDBDatabase} */
+                const db = request.result;
+
+                if (event.oldVersion < 1) {
+                    // Version 1 is the first version of the database.
+                    const store = db.createObjectStore(KeyStore.DB_KEY_STORE_NAME, { keyPath: 'userFriendlyAddress' });
```

Comment:
no auto incremented id then?

Reply:
We decided against adding another identifier, because we only have the one-key-to-one-address relation currently. Future keys/addresses that are derived from seed phrases will be stored differently.